### PR TITLE
Prepare for v0.8.0 release

### PR DESCRIPTION
**Pull request by llucax**
Bump the version in the cookiecutter template and update the release notes.

### CODE REVIEW

**Comment by llucax** [comment]
This one is to merge last, and will probably need to fix some conflicts, so setting as draft. But the contents can be reviewed now.

**Comment by llucax** [comment]
Rebased and set as ready for review.

**Comment by llucax** [comment]
Sorry,  I forgot to add upgrading instructions, will update soon again.

**Comment by llucax** [comment]
Done!

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -2,15 +2,33 @@
...
-<!-- Here goes notes on how to upgrade from previous versions, including deprecations and what they should be replaced with -->
-
 ### Cookiecutter template
 
-<!-- Here upgrade steps for cookiecutter specifically -->
+You can follow these steps to upgrade without regenerating the whole project, if you kept the default project structure and configuration:
+
+* Update the `frequenz-repo-config` dependencies in `pyproject.toml` to 0.8.0.
+
+* Run in the root of your project:
```
Comment (daniel-zullo-frequenz) [inline RELEASE_NOTES.md]: What about clarify explicitly that is the root directory?
```suggestion
* Run in the root directory of your project:
```

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -2,15 +2,33 @@
...
+
+    # Auto-generated python files from the protocol buffer compiler
+    py/**/*_pb2.py
+    py/**/*_pb2.pyi
+    py/**/*_pb2_grpc.py
+    py/**/*_pb2_grpc.pyi
+    EOF
+    ```
+
+* Optionally go to the project's settings and replace branch protection rules with the new rulesets. See the new [GitHub configuration guide](https://frequenz-floss.github.io/frequenz-repo-config-python/v0.8/user-guide/start-a-new-project/github-configuration/#branches) for more details.
```
Comment (daniel-zullo-frequenz) [inline RELEASE_NOTES.md]:
```suggestion
* Optionally go to the GitHub project's settings and replace branch protection rules with the new rulesets. See the new [GitHub configuration guide](https://frequenz-floss.github.io/frequenz-repo-config-python/v0.8/user-guide/start-a-new-project/github-configuration/#branches) for more details.
```

**Review by daniel-zullo-frequenz — DISMISSED** [review]
I have 2 optional suggestions, LGTM

**Comment by daniel-zullo-frequenz** [comment]
Sorry, I removed it from the merging queue in case you'd like to address the optional suggestions

**Review by daniel-zullo-frequenz — DISMISSED** [review]

**Comment by llucax** [comment]
Updated with the suggestions and also updated the tests (they will probably still fail because the version in the dependencies doesn't exist yet, this can be only fixed when we split this repo).

**Review by daniel-zullo-frequenz — APPROVED** [review]

**Pull request merged**